cloudsync/youtube.py

```diff
@@ -0,0 +1,247 @@
+""" YouTube API interface"""
+import time
+import http
+import logging
+from tempfile import NamedTemporaryFile
+
+import boto3
+import httplib2
+import oauth2client
+
+from django.conf import settings
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaFileUpload
+
+from ui.encodings import EncodingNames
+
+log = logging.getLogger(__name__)
+
+
+def resumable_upload(request):
+    """
+    Upload a video to YouTube and resume on failure up to 10 times, adapted from YouTube API example
+    """
+    response = None
+    error = None
+    retry = 0
+    max_retries = 10
+
+    retry_exceptions = (IOError, httplib2.HttpLib2Error, http.client.NotConnected,
+                        http.client.IncompleteRead, http.client.ImproperConnectionState,
+                        http.client.CannotSendRequest, http.client.CannotSendHeader,
+                        http.client.ResponseNotReady, http.client.BadStatusLine)
+    retry_statuses = [500, 502, 503, 504]
```

Comment:
Where do these statuses and exceptions come from?

Reply:
Google/Youtube API